File: backend/src/models/RssFeed.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const RssFeed = sequelize.define('RssFeed', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  name: {
    type: DataTypes.STRING(255),
    allowNull: false,
    comment: 'RSS源名称'
  },
  url: {
    type: DataTypes.TEXT,
    allowNull: false,
    comment: 'RSS源URL'
  },
  category: {
    type: DataTypes.STRING(100),
    allowNull: true,
    comment: '分类标签'
  },
  description: {
    type: DataTypes.TEXT,
    allowNull: true,
    comment: '描述'
  },
  isActive: {
    type: DataTypes.BOOLEAN,
    defaultValue: true,
    comment: '是否启用'
  },
  lastFetchTime: {
    type: DataTypes.DATE,
    allowNull: true,
    comment: '最后抓取时间'
  },
  fetchInterval: {
    type: DataTypes.INTEGER,
    defaultValue: 3600000, // 1小时
    comment: '抓取间隔(毫秒)'
  }
}, {
  tableName: 'rss_feeds',
  timestamps: true,
  indexes: [
    {
      fields: ['category']
    },
    {
      fields: ['isActive']
    }
  ]
});

module.exports = RssFeed; 

File: backend-go/internal/models/rss_feed.go
package models

import (
	"time"
)

type RssFeed struct {
	ID           uint       `gorm:"primaryKey" json:"id"`
	Name         string     `gorm:"column:name;size:255;not null" json:"name"`
	URL          string     `gorm:"column:url;type:text;not null" json:"url"`
	Category     string     `gorm:"column:category;size:100;default:'其他'" json:"category"`
	Description  string     `gorm:"column:description;type:text" json:"description"`
	IsActive     bool       `gorm:"column:isActive;default:true" json:"isActive"`
	LastFetchTime *time.Time `gorm:"column:lastFetchTime" json:"lastFetchTime"`
	FetchInterval int64      `gorm:"column:fetchInterval;default:3600000" json:"fetchInterval"`
	ArticleCount int         `gorm:"-" json:"articleCount"`
	CreatedAt    time.Time  `gorm:"column:createdAt" json:"createdAt"`
	UpdatedAt    time.Time  `gorm:"column:updatedAt" json:"updatedAt"`
}

func (RssFeed) TableName() string {
	return "rss_feeds"
}
